libfs/file_info.go

```diff
@@ -82,39 +88,62 @@ type fileInfoSys struct {
 	fi *FileInfo
 }
 
-var _ LastWriterGetter = fileInfoSys{}
+type KBFSMetadataForSimpleFS struct {
+	LastWriter     keybase1.User
+	PrefetchStatus keybase1.PrefetchStatus
+}
+
+var _ KBFSMetadataForSimpleFSGetter = fileInfoSys{}
 
-func (fis fileInfoSys) LastWriter() (keybase1.User, error) {
+func (fis fileInfoSys) KBFSMetadataForSimpleFS() (KBFSMetadataForSimpleFS,
+	error) {
 	if fis.fi.node == nil {
 		// This won't return any last writer for symlinks themselves.
 		// TODO: if we want symlink last writers, we'll need to add a
 		// new interface to KBFSOps to get them.
-		return keybase1.User{}, nil
+		return KBFSMetadataForSimpleFS{}, nil
 	}
 	md, err := fis.fi.fs.config.KBFSOps().GetNodeMetadata(
 		fis.fi.fs.ctx, fis.fi.node)
 	if err != nil {
-		return keybase1.User{}, err
+		return KBFSMetadataForSimpleFS{}, err
 	}
+	var prefetchStatus keybase1.PrefetchStatus
+	switch md.PrefetchStatus {
+	case "NoPrefetch":
```

Comment:
I wonder if we should refactor `NodeMetadata` to contain a `PrefetchStatus` instead of a string. Not a big deal though.

Reply:
Oddly, as far as I can tell, there's no existing references to prefetch status on the node metadata? Which would make that change very straightforward. Though it might cause confusion w/r/t the *other* prefetch status struct (in `data_types.go`). Maybe we should just make everything be `keybase1.PrefetchStatus`?

Reply:
Existing references are when the `NodeMetadata` is output directly  in `cat /keybase/{private,public,team}/tlf/.kbfs_fileinfo_foo.txt`. But if you change it to a `libkbfs.PrefetchStatus`, it'll inherit its `String()` method, which should hopefully also output nicely.

Reply:
I'd prefer to avoid using `keybase1.PrefetchStatus` in `libkbfs` code since it's string-based instead of integer-based.